components/canvas/canvas_paint_thread.rs

```diff
@@ -864,3 +864,180 @@ impl ToAzFloat for Rect<f64> {
                   Size2D::new(self.size.width as AzFloat, self.size.height as AzFloat))
     }
 }
+
+
+pub trait ToAzureStyle {
+    type Target;
+    fn to_azure_style(self) -> Self::Target;
+}
+
+impl ToAzureStyle for FillOrStrokeStyle {
+    type Target = Option<Pattern>
+    fn to_azure_pattern(&self, drawtarget: &DrawTarget) -> Self::Target {
```

Comment:
Have you tried compiling your changes? This does not(look like it should compile.